sktime/forecasting/statsforecast.py

```diff
@@ -14,58 +14,104 @@
 _check_soft_dependencies("statsforecast", severity="warning")
 
 
-class AutoARIMA(_StatsForecastAdapter):
+class StatsForecastAutoARIMA(_StatsForecastAdapter):
```

Comment:
why do we actually have this class `_StatsForecastAdapter`? Do you plan to add more models that inherit from it? To have a parent for only one child makes no sense to me here otherwise

Reply:
I feel it might be worth factoring this out, there seem to be interface commonalities with the `PmdarimaAdapter` which could be unified in the next roud of refactor. E.g., the entire insample/out-of-sample case distinction business.

Reply:
Yes, @aiwalter. In the future, I would like to add more statsforecast models :) and I agree with @fkiraly, I used the `PmdarimaAdaptar` as initial reference but in future iterations, I plan to refactor it.